ccl/changefeedccl: TestChangefeedTimestamps failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/d57a1f0f985b95226e38f20a2afa006b9e48e92e

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stressrace instead of stress and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
make stress TESTS=TestChangefeedTimestamps PKG=github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl TESTTIMEOUT=5m STRESSFLAGS='-stderr=false -maxtime 20m -timeout 10m'
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=868061&tab=buildLog

```
=== RUN   TestChangefeedTimestamps
W180829 05:07:14.486975 2892 server/status/runtime.go:310  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I180829 05:07:14.493229 2892 server/server.go:827  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I180829 05:07:14.493436 2892 base/addr_validation.go:271  [n?] server certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I180829 05:07:14.493487 2892 base/addr_validation.go:311  [n?] web UI certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I180829 05:07:14.513863 2892 server/config.go:493  [n?] 1 storage engine initialized
I180829 05:07:14.513888 2892 server/config.go:496  [n?] RocksDB cache size: 128 MiB
I180829 05:07:14.513897 2892 server/config.go:496  [n?] store 0: in-memory, size 0 B
I180829 05:07:14.535237 2892 server/node.go:373  [n?] **** cluster b06fb2a5-27db-4503-8734-8823f3125677 has been created
I180829 05:07:14.535270 2892 server/server.go:1389  [n?] **** add additional nodes by specifying --join=127.0.0.1:42009
I180829 05:07:14.535536 2892 gossip/gossip.go:382  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:42009" > attrs:<> locality:<> ServerVersion:<major_val:2 minor_val:0 patch:0 unstable:12 > build_tag:"v2.1.0-alpha.20180702-2098-gd57a1f0" started_at:1535519234535454257 
I180829 05:07:14.541664 2892 server/node.go:476  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=6.9 KiB), ranges=1, leases=1, queries=0.00, writes=0.00, bytesPerReplica={p10=7103.00 p25=7103.00 p50=7103.00 p75=7103.00 p90=7103.00 pMax=7103.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
I180829 05:07:14.541757 2892 storage/stores.go:242  [n1] read 0 node addresses from persistent storage
I180829 05:07:14.541944 2892 server/node.go:697  [n1] connecting to gossip network to verify cluster ID...
I180829 05:07:14.541988 2892 server/node.go:722  [n1] node connected via gossip and verified as part of cluster "b06fb2a5-27db-4503-8734-8823f3125677"
I180829 05:07:14.542016 2892 server/node.go:546  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I180829 05:07:14.542483 2892 server/status/recorder.go:652  [n1] available memory from cgroups (8.0 EiB) exceeds system memory 16 GiB, using system memory
I180829 05:07:14.542511 2892 server/server.go:1780  [n1] Could not start heap profiler worker due to: directory to store profiles could not be determined
I180829 05:07:14.542596 2892 server/server.go:1526  [n1] starting https server at 127.0.0.1:38591 (use: 127.0.0.1:38591)
I180829 05:07:14.548862 2892 server/server.go:1528  [n1] starting grpc/postgres server at 127.0.0.1:42009
I180829 05:07:14.543284 3162 server/status/recorder.go:652  [n1,summaries] available memory from cgroups (8.0 EiB) exceeds system memory 16 GiB, using system memory
I180829 05:07:14.548935 2892 server/server.go:1529  [n1] advertising CockroachDB node at 127.0.0.1:42009
I180829 05:07:14.563455 3174 sql/event_log.go:126  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I180829 05:07:14.587669 2846 sql/event_log.go:126  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:2.0-12 User:root}
I180829 05:07:14.593006 3147 storage/replica_command.go:298  [split,n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I180829 05:07:14.601330 3187 storage/replica_command.go:298  [split,n1,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/NodeLiveness [r3]
I180829 05:07:14.602723 3131 sql/event_log.go:126  [n1,intExec=disableNetTrace] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:root}
I180829 05:07:14.622059 3216 storage/replica_command.go:298  [split,n1,s1,r3/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/NodeLivenessMax [r4]
I180829 05:07:14.625515 3211 sql/event_log.go:126  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:bf603305-cd16-435d-8a64-d116dd1815ab User:root}
I180829 05:07:14.633877 3190 sql/event_log.go:126  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I180829 05:07:14.648235 3185 storage/replica_command.go:298  [split,n1,s1,r4/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/tsd [r5]
I180829 05:07:14.652758 3194 sql/event_log.go:126  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I180829 05:07:14.654652 3237 storage/replica_command.go:298  [split,n1,s1,r5/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r6]
I180829 05:07:14.662374 3227 storage/replica_command.go:298  [split,n1,s1,r6/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r7]
I180829 05:07:14.663710 2892 server/server.go:1582  [n1] done ensuring all necessary migrations have run
I180829 05:07:14.663742 2892 server/server.go:1585  [n1] serving sql connections
I180829 05:07:14.672244 3251 server/server_update.go:67  [n1] no need to upgrade, cluster already at the newest version
I180829 05:07:14.673229 3253 sql/event_log.go:126  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:42009} Attrs: Locality: ServerVersion:2.0-12 BuildTag:v2.1.0-alpha.20180702-2098-gd57a1f0 StartedAt:1535519234535454257 LocalityAddress:[]} ClusterID:b06fb2a5-27db-4503-8734-8823f3125677 StartedAt:1535519234535454257 LastUp:1535519234535454257}
I180829 05:07:14.675153 3283 storage/replica_command.go:298  [split,n1,s1,r7/1:/{Table/System…-Max}] initiating a split of this range at key /Table/11 [r8]
I180829 05:07:14.688338 3231 storage/replica_command.go:298  [split,n1,s1,r8/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r9]
I180829 05:07:14.693699 3309 storage/replica_command.go:298  [split,n1,s1,r9/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r10]
I180829 05:07:14.699481 3331 storage/replica_command.go:298  [split,n1,s1,r10/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r11]
I180829 05:07:14.705485 1297 storage/replica_command.go:298  [split,n1,s1,r11/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r12]
I180829 05:07:14.709206 3371 storage/replica_command.go:298  [split,n1,s1,r12/1:/{Table/15-Max}] initiating a split of this range at key /Table/16 [r13]
I180829 05:07:14.714243 3324 storage/replica_command.go:298  [split,n1,s1,r13/1:/{Table/16-Max}] initiating a split of this range at key /Table/17 [r14]
I180829 05:07:14.719230 3342 storage/replica_command.go:298  [split,n1,s1,r14/1:/{Table/17-Max}] initiating a split of this range at key /Table/18 [r15]
I180829 05:07:14.725052 3422 storage/replica_command.go:298  [split,n1,s1,r15/1:/{Table/18-Max}] initiating a split of this range at key /Table/19 [r16]
I180829 05:07:14.732977 3400 storage/replica_command.go:298  [split,n1,s1,r16/1:/{Table/19-Max}] initiating a split of this range at key /Table/20 [r17]
I180829 05:07:14.749189 3347 storage/replica_command.go:298  [split,n1,s1,r17/1:/{Table/20-Max}] initiating a split of this range at key /Table/21 [r18]
W180829 05:07:14.762066 3446 storage/intent_resolver.go:668  [n1,s1] failed to push during intent resolution: failed to push "split" id=38e058c9 key=/Local/Range/Table/20/RangeDescriptor rw=true pri=0.00774032 iso=SERIALIZABLE stat=PENDING epo=0 ts=1535519234.749200239,0 orig=1535519234.749200239,0 max=1535519234.749200239,0 wto=false rop=false seq=1
I180829 05:07:14.763380 3350 storage/replica_command.go:298  [split,n1,s1,r18/1:/{Table/21-Max}] initiating a split of this range at key /Table/22 [r19]
I180829 05:07:14.768227 3453 storage/replica_command.go:298  [split,n1,s1,r19/1:/{Table/22-Max}] initiating a split of this range at key /Table/23 [r20]
I180829 05:07:14.775988 3408 storage/replica_command.go:298  [split,n1,s1,r20/1:/{Table/23-Max}] initiating a split of this range at key /Table/50 [r21]
I180829 05:07:14.793712 3490 storage/replica_command.go:298  [split,n1,s1,r21/1:/{Table/50-Max}] initiating a split of this range at key /Table/51 [r22]
I180829 05:07:14.861728 3508 sql/event_log.go:126  [n1,client=127.0.0.1:34190,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:changefeed.experimental_poll_interval Value:0s User:root}
I180829 05:07:14.866993 3508 sql/event_log.go:126  [n1,client=127.0.0.1:34190,user=root] Event: "create_database", target: 52, info: {DatabaseName:d Statement:CREATE DATABASE d User:root}
I180829 05:07:14.867504 3501 storage/replica_command.go:298  [split,n1,s1,r22/1:/{Table/51-Max}] initiating a split of this range at key /Table/52 [r23]
I180829 05:07:14.874898 3508 sql/event_log.go:126  [n1,client=127.0.0.1:34190,user=root] Event: "create_table", target: 53, info: {TableName:d.public.foo Statement:CREATE TABLE foo (a INT PRIMARY KEY) User:root}
I180829 05:07:14.877624 3559 storage/replica_command.go:298  [split,n1,s1,r23/1:/{Table/52-Max}] initiating a split of this range at key /Table/53 [r24]
--- FAIL: TestChangefeedTimestamps (0.48s)
	changefeed_test.go:188: error scanning '&{<nil> 0xc42162e080}': pq: restart transaction: HandledRetryableTxnError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE): "sql txn" id=caf3565b key=/Table/53/1/1/0 rw=true pri=0.00518323 iso=SERIALIZABLE stat=PENDING epo=0 ts=1535519234.919199227,1 orig=1535519234.918512794,0 max=1535519234.918512794,0 wto=false rop=false seq=3

```

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -20,6 +20,8 @@ import (
 	"sync"
 	"testing"
 
+	"github.com/cockroachdb/cockroach-go/crdb"
+
 	"github.com/cockroachdb/apd"
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
@@ -175,17 +177,25 @@ func TestChangefeedTimestamps(t *testing.T) {
 	sqlDB.Exec(t, `CREATE TABLE foo (a INT PRIMARY KEY)`)
 
 	var ts0 string
-	sqlDB.QueryRow(t,
-		`BEGIN; INSERT INTO foo VALUES (0); SELECT cluster_logical_timestamp(); COMMIT`,
-	).Scan(&ts0)
+	if err := crdb.ExecuteTx(ctx, sqlDB.DB, nil /* txopts */, func(tx *gosql.Tx) error {
+		return tx.QueryRow(
+			`INSERT INTO foo VALUES (0) RETURNING cluster_logical_timestamp()`,
+		).Scan(&ts0)
+	}); err != nil {
+		t.Fatal(err)
+	}
 
 	rows := sqlDB.Query(t, `CREATE CHANGEFEED FOR foo WITH updated, resolved`)
 	defer closeFeedRowsHack(t, sqlDB, rows)
 
 	var ts1 string
-	sqlDB.QueryRow(t,
-		`BEGIN; INSERT INTO foo VALUES (1); SELECT cluster_logical_timestamp(); COMMIT`,
-	).Scan(&ts1)
+	if err := crdb.ExecuteTx(ctx, sqlDB.DB, nil /* txopts */, func(tx *gosql.Tx) error {
+		return tx.QueryRow(
+			`INSERT INTO foo VALUES (1) RETURNING cluster_logical_timestamp()`,
+		).Scan(&ts1)
+	}); err != nil {
+		t.Fatal(err)
+	}
 
 	assertPayloads(t, rows, []string{
 		`foo: [0]->{"__crdb__": {"updated": "` + ts0 + `"}, "a": 0}`,

